gcs_contents_manager.py

```diff
@@ -428,3 +449,262 @@ def rename_file(self, old_path, new_path):
+    return checkpoint_manager.create_file_checkpoint(content, format, relative_path)
+
+  def create_notebook_checkpoint(self, nb, path):
+    checkpoint_manager, relative_path = self._checkpoint_manager_for_path(path)
+    if not checkpoint_manager:
+      raise HTTPError(500, 'Unsupported checkpoint path: {}'.format(path))
+    return checkpoint_manager.create_notebook_checkpoint(nb, relative_path)
+
+  def get_file_checkpoint(self, checkpoint_id, path):
+    checkpoint_manager, relative_path = self._checkpoint_manager_for_path(path)
+    if not checkpoint_manager:
+      raise HTTPError(500, 'Unsupported checkpoint path: {}'.format(path))
+    return checkpoint_manager.get_file_checkpoint(checkpoint_id, relative_path)
+
+  def get_notebook_checkpoint(self, checkpoint_id, path):
+    checkpoint_manager, relative_path = self._checkpoint_manager_for_path(path)
+    if not checkpoint_manager:
+      raise HTTPError(500, 'Unsupported checkpoint path: {}'.format(path))
+    return checkpoint_manager.get_notebook_checkpoint(checkpoint_id, relative_path)
+
+  def delete_checkpoint(self, checkpoint_id, path):
+    checkpoint_manager, relative_path = self._checkpoint_manager_for_path(path)
+    if not checkpoint_manager:
+      raise HTTPError(500, 'Unsupported checkpoint path: {}'.format(path))
+    return checkpoint_manager.delete_checkpoint(checkpoint_id, relative_path)
+
+  def list_checkpoints(self, path):
+    checkpoint_manager, relative_path = self._checkpoint_manager_for_path(path)
+    if not checkpoint_manager:
+      raise HTTPError(500, 'Unsupported checkpoint path: {}'.format(path))
+    return checkpoint_manager.list_checkpoints(relative_path)
+
+  def rename_checkpoint(self, checkpoint_id, old_path, new_path):
+    checkpoint_manager, old_relative_path = self._checkpoint_manager_for_path(old_path)
+    if not checkpoint_manager:
+      raise HTTPError(500, 'Unsupported checkpoint path: {}'.format(old_path))
+    new_checkpoint_manager, new_relative_path = self._checkpoint_manager_for_path(new_path)
+    if new_checkpoint_manager != checkpoint_manager:
+      raise HTTPError(500, 'Unsupported rename: {}->{}'.format(old_path, new_path))
+    return checkpoint_manager.rename_checkpoint(checkpoint_id, old_relative_path, new_relative_path)
+
+
+class CombinedContentsManager(ContentsManager):
+  root_dir = Unicode(config=True)
+
+  @default('checkpoints')
+  def _default_checkpoints(self):
+    return CombinedCheckpointsManager(self._content_managers)
+
+  def __init__(self, **kwargs):
+    print('Creating the combined contents manager...')
+    super(CombinedContentsManager, self).__init__(**kwargs)
+
+    file_cm = FileContentsManager(**kwargs)
+    file_cm.checkpoints = GenericFileCheckpoints(**file_cm.checkpoints_kwargs)
+    gcs_cm = GCSContentsManager(**kwargs)
+    self._content_managers = {
+      'Local Disk': file_cm,
+      'GCS': gcs_cm,
+    }
+
+  def _content_manager(self, path):
```

Comment:
Nit: _content_manager_for_path for consistency with corresponding CheckpointManager method above?

Reply:
Done